### PR TITLE
doc: upgrade mongo-driver

### PR DESCRIPTION
**Pull request by phiz71**
update gravitee.yml default configuration for mongo database

gravitee-io/issues#3661

### CODE REVIEW

**Review by gaetanmaisse — COMMENTED** [review]
I think we should add something about the attributes that aren't handled anymore: 
https://github.com/gravitee-io/gravitee-api-management/blob/72e88c3b229aa7ed949989708e4711d51ea31b21/gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/common/MongoFactory.java#L180

**Review by gaetanmaisse — APPROVED** [review]

File: pages/apim/3.x/installation-guide/configuration/repositories/installation-guide-repositories-mongodb.adoc [hunk 1]
```diff
@@ -53,31 +53,63 @@ management:
...
+    writeConcern:               # possible values are 1,2,3... (the number of node) or 'majority' (default is 1)
+    wtimeout:                   # (default is 0)
+    journal:                    # (default is true)
+
+## Socket settings
+    connectTimeOut:             # mongodb connection time out (default 1000)
+    socketTimeout:              # mongodb socket time out (default 1000)
+
+## Cluster settings
+    serverSelectionTimeout:     # mongodb server selection time out (default 1000)
```
Comment (tiffanymarchi) [inline pages/apim/3.x/installation-guide/configuration/repositories/installation-guide-repositories-mongodb.adoc:74]: Could we please change "time out" to "timeout" in the definition for all of these? :) 

## Socket settings
    connectTimeOut:             # mongodb connection timeout (default 1000)
    socketTimeout:              # mongodb socket timeout (default 1000)

File: pages/apim/3.x/installation-guide/configuration/repositories/installation-guide-repositories-mongodb.adoc [hunk 2]
```diff
@@ -53,31 +53,63 @@ management:
...
-    threadsAllowedToBlockForConnectionMultiplier: # mongodb threads allowed to block for connection multiplier (default null)
-    cursorFinalizerEnabled:     # mongodb cursor finalizer enabled (default false)
+    keystore:                   # path to KeyStore (when sslEnabled is true, default null)
+    keystorePassword:           # KeyStore password (when sslEnabled is true, default null)
+    keyPassword:                # password for recovering keys in the KeyStore (when sslEnabled is true, default null)
 ----
 
+[WARNING]
+====
+From version 3.10.0, Gravitee.io APIM uses the 4.1.2 version of the Java Driver. As a consequence, some settings are not available anymore.
```
Comment (tiffanymarchi) [inline pages/apim/3.x/installation-guide/configuration/repositories/installation-guide-repositories-mongodb.adoc:97]: Let's please change: "As a consequence, some settings are not available anymore."

To: Therefore, some settings are no longer available.

**Review by tiffanymarchi — CHANGES_REQUESTED** [review]
Overall, very nicely done, Florent! I just have a couple minor edits. Thank you!  (●'◡'●)

**Review by tiffanymarchi — APPROVED** [review]

**Pull request merged**